### PR TITLE
[risk=low][no ticket] Stop referencing AdminTableUser.rasLinkLoginGovBypassTime

### PR DESCRIPTION
**Pull request by jmthibault79**
Followup to #8362 - since this field is deprecated, we should stop using it.  As a first step, the UI can stop using it.

Tested locally by doing some bypassing

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java [hunk 1]
```diff
@@ -129,7 +128,7 @@ public List<AccessModuleStatus> getAccessModuleStatus(DbUser user) {
         .map(this::maybeReturnAccessModule)
         .filter(Optional::isPresent)
         .map(Optional::get)
-        .collect(Collectors.toList());
+        .toList();
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:131]: the changes in this file are just refactoring to use more recent Java features, as I happened to be looking at it

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java [hunk 2]
```diff
@@ -252,22 +251,15 @@ public static Timestamp deriveExpirationTimestamp(Timestamp completionTime, Long
...
-        return accessConfig.enableRasLoginGovLinking;
-      case RAS_LINK_ID_ME:
-        return accessConfig.enableRasIdMeLinking;
-      default:
-        return true;
-    }
+      case COMPLIANCE_TRAINING, CT_COMPLIANCE_TRAINING -> accessConfig.enableComplianceTraining;
+      case RAS_LINK_LOGIN_GOV -> accessConfig.enableRasLoginGovLinking;
+      case RAS_LINK_ID_ME -> accessConfig.enableRasIdMeLinking;
+      case IDENTITY -> accessConfig.enableRasLoginGovLinking || accessConfig.enableRasIdMeLinking;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:261]: This code change should not cause any functional changes given that these 2 values happen to always be equal.  However: this better expresses when we want Identity to be required.

File: ui/src/app/pages/admin/user/admin-user-table.tsx [hunk 3]
```diff
@@ -231,7 +231,7 @@ export const AdminUserTable = withUserProfile()(
           dataAccess: this.dataAccessContents(user),
           bypass: (
             <AdminUserBypass
-              user={{ ...user }}
+              {...{ user }}
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/user/admin-user-table.tsx:234]: unrelated cleanup - I'm sure this is what was intended

File: ui/src/app/pages/admin/user/admin-user-bypass.tsx [hunk 4]
```diff
@@ -45,9 +45,6 @@ const getBypassedModules = (user: AdminTableUser): Array<AccessModule> => {
     ...(user.duccBypassTime ? [AccessModule.DATA_USER_CODE_OF_CONDUCT] : []),
     ...(user.eraCommonsBypassTime ? [AccessModule.ERA_COMMONS] : []),
     ...(user.twoFactorAuthBypassTime ? [AccessModule.TWO_FACTOR_AUTH] : []),
-    ...(user.rasLinkLoginGovBypassTime
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/user/admin-user-bypass.tsx:48]: this was not being used

File: ui/src/testing/stubs/profile-api-stub.ts [hunk 5]
```diff
@@ -25,11 +25,6 @@ export class ProfileStubVariables {
     areaOfResearch: 'things',
     generalDiscoverySources: [],
     partnerDiscoverySources: [],
-    complianceTrainingCompletionTime: null,
```
Comment (jmthibault79) [inline ui/src/testing/stubs/profile-api-stub.ts:28]: these fields haven't been in Profile for a long time

**Review by evrii — APPROVED** [review]
Tested locally by navigating to the admin user list and looking at bypass actions for a user. I also opened an individual user's admin page. All this worked without issue.

Thanks for cleaning this up.
